Save flash - use DEBUG for SERIAL (UBL)

classicrocker883 opened the pull request:
<!--

Submitting a Pull Request

- Please fill out all sections of this form. You can delete the helpful comments.
- Pull Requests without clear information will take longer and may even be rejected.
- We get a high volume of submissions so please be patient during review.

-->

### Description
~~Changing `SERIAL_ECHO*` to `DEBUG_ECHO*`~~
Add option to disable serial output for UBL in **ubl_G29.cpp** results in saving ~1976 bytes (previously 1432) using new `DISABLE_UBL_SERIAL_CHITCHAT`

- Also adds `DEBUG_ECHO_SP` to **debug_out.h**
- add `weight_factor` to `static void smart_fill_wlsf(const float )` and
`TERN(AVR_ATmega2560_FAMILY, __O2, __Os); // O2 gives smaller code than Os on A2560`

~~Probably some serial echo could be kept, after all `DEBUG_LEVELING_FEATURE` *is* "**detailed logging of G28, G29, M48, etc.**".~~
<!--

Clearly describe the submitted changes with lots of details. Include images where helpful. Initial reviewers may not be familiar with the subject, so be as thorough as possible. You can use MarkDown syntax to improve readability with bullet lists, code blocks, and so on. PREVIEW and fix up formatting before submitting.

-->
>[!TIP]
>Thoughts?
>Comments?
>Suggestions?
>Changes?

### Requirements

<!-- Does this PR require a specific board, LCD, etc.? -->

### Benefits
- Keep original integrity
- Save flash if needed
- Fix/improve on `DEBUG_LEVELING_FEATURE` / `VALIDATE_MESH_TILT`
<!-- What does this PR fix or improve? -->

### Configurations

<!-- Attach Configurations ZIP and any other files needed to test this PR. -->

### Related Issues
there are multiple errors when `VALIDATE_MESH_TILT` is enabled (in **ubl_g29.cpp)** - such as `gotz` and `probe_pt` are undefined.  
this **PR** does fix that by rearranging the values
<!-- Does this PR fix a bug or fulfill a Feature Request? Link related Issues here. -->

classicrocker883 committed:
use debug for serial - save memory

thinkyhead committed:
Merge branch 'bugfix-2.1.x' into pr/28184

ellensp commented:
Many advanced users do not use displays, they are far to limiting.
This PR looks like it makes normal command line use of ubl_G29 without a screen impossible.   
Many of these are not debugging they are basic operations for serial gcode operation.

Now if you where to add a feature to advanced eg "DISABLE_UBL_SERIAL_CHITCHAT"  or something like that I think this would be a better option than redefining it all as debugging. the user gets to choose if they want it or not.

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
adjust debug leveling - fix `normal`

classicrocker883 committed:
fix, revert DEBUG_OUT

classicrocker883 committed:
switch to `DISABLE_UBL_SERIAL_CHITCHAT`

classicrocker883 committed:
fix/rearrange `VALIDATE_MESH_TILT`

classicrocker883 committed:
revert position

classicrocker883 committed:
better, revert back to using DEBUG_ECHO

classicrocker883 committed:
revert oldRpos

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
ubl_G29 fixes

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-November3

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
add `TERN` to `smart_fill_wlsf`

Used AVR_ATmega2560_FAMILY - maybe instead of AVR_ATmega2560?
// O2 gives smaller code than Os on A2560

classicrocker883 committed:
add parameter name for clarity

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' into bugfix-2.1.x-November3
